src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContext.cs

```diff
@@ -41,6 +41,17 @@ internal interface IWorkspaceProjectContext : IDisposable
         void RemoveAdditionalFile(string filePath);
         void AddDynamicFile(string filePath, IEnumerable<string> folderNames = null);
         void RemoveDynamicFile(string filePath);
+
+        /// <summary>
+        /// Adds a file (like a .editorconfig) used to configure analyzers.
+        /// </summary>
+        void AddAnalyzerOptionsFile(string filePath);
```

Comment:
1. How to we know the path of editorconfig file today? using additional files?
2. Is this  intended for something more general, or just editorconfig?

Reply:
Right now it's discovered by the IDE at runtime with no involvement of the project system. That's only because we haven't done any integration with the build system, but we're doing that now.

Regarding naming, we're following @agocke's naming convention that rather than hard coding "editorconfig" into the name, we're calling them "analyzer options files", so if we one day are supporting another syntax we're not renaming everything.

Reply:
Actually sorry I guess the name is Analyzer*Config*Options or the like. Will update this.

Reply:
Yeah the command line switch is `/analyzerconfig` so perhaps go with `AddAnalyzerConfigFile` and `RemoveAnalyzerConfigFile`? I'm updating my project system code to match.

Reply:
Yup, my base name for EditorConfig is "AnalyzerConfig"

Reply:
OK, so you have to bear with me since I don't know much about editorconfig, but if `/analyzerconfig` is a compiler argument, why it is not part of `CompilationOptions` instead?